### ISSUE TITLE
Problems with Aqara curtains since last update

### ISSUE THREAD

**Issue by kmarosi** [swttt/com.swttt.homekit#190]
Hi,

Before the v3.0.16 of Homeykit my Aqara curtains worked perfectly fine with Homey and in Homekit through the Homeykit app. But since then, I can't control those curtains through Homekit. The devices show up in Homekit but nothing happens when I change open/close.

I went into developer.athom.com and check what happens. When I change status of the curtains in Homekit the windowcoverings_set value changes but nothing happens. When I change status of the curtains in Homey the dim value changes and the curtains start moving.

Any ideas what this could be?

**Comment by robertklep** [comment]
Which specific app are you using to control the curtains? 3.0.16 contains some changes to how curtains are being handled (to fix problems with _other_ apps), so I have to check and see how the Aqara app handles them.

**Comment by kmarosi** [comment]
I use the Aqara & Xiaomi Zigbee (Version 0.8.0) app.

**Comment by robertklep** [comment]
For some reason, the driver add the `windowcoverings_set` capability to the device, but doesn't use actually use it. This confuses Homeykit.

I'll ask @tedtolboom about it.

**Comment by robertklep** [comment]
@kmarosi I added a workaround, can you check if this fixes your problem? https://homey.app/a/com.swttt.homekit/test/

**Comment by kmarosi** [comment]
@robertklep - I managed to close them but then some weird things started to happen. The status changed to Closing... and they closed but then the status changed back to Open xx%. And after that the curtains started to open and close by itself for a few times. Curtains are now open but the status is Closed.

**Comment by robertklep** [comment]
@kmarosi I don't really see how Homeykit can cause that sort of behaviour. Do you know which version were running previous to 3.0.16? Do you know how to use the CLI method to install apps on Homey?

**Comment by kmarosi** [comment]
@robertklep I've been running Homeykit since 3.0.11. The last version where everything worked as it should was with 3.0.15, so my problems started a few days ago.

I haven't installed apps with the CLI method on Homey before, but that's something I should be able to do. Just let me know what commands you want me try. :)

**Comment by robertklep** [comment]
@kmarosi instructions on how to install an app through the CLI are here:
https://community.athom.com/t/how-to-cli-install-method/198

Just to make sure that 3.0.15 still works for you, go to [this URL](https://github.com/swttt/com.swttt.homekit/tree/988c4c797feb82df998549dc377562e76717c00a), click the green button "Clone or download", then "Download ZIP".

Unpack the ZIP file, which should unpack to a directory called "com.swttt.homekit-988c4c797feb82df998549dc377562e76717c00a". Go to that directory, and run `homey app run`

That will run (but not install) v3.0.15. Check and see if that still works as expected.

**Comment by kmarosi** [comment]
@robertklep Interestingly enough, it doesn't work with 3.0.15 either anymore.. Closing works fine. It first goes from Closing.. then to Closed. But when I try to open them the status changes to Opening... and then it starts opening/closing after it eventually closes itself.

Could it be a Zigbee error with the Homey?

**Comment by robertklep** [comment]
@kmarosi start by taking Homey off power for about 10 minutes and see if it works again after that. I assume that the device works okay when you control it directly from the Homey app (taking Homeykit out of the equation)?

**Comment by kmarosi** [comment]
@robertklep I'll try that! I'll get back to you shortly. Works perfectly fine when controlling it directly from Homey. No flows that are supposed to be in the way, but I'll disable any flow that has the curtains involved just to make sure 100%.

**Comment by kmarosi** [comment]
@robertklep Unfortunately that didn't do the trick either. It sorted one problem though, it doesn't open/close back and forth now. The problem now is that the status is not changing after Opening... and Closing... and if I restart the Home app it goes back to the previous setting, even though it actually opened or closed the curtain.

**Comment by robertklep** [comment]
Are you still testing with 3.0.15?

**Comment by kmarosi** [comment]
@robertklep I didn’t think of that. But when I check the Homeykit app it says it runs 3.0.15 even after the reboot. Is that wrong or should I do another reboot, run 3.0.15 again manually and then test?

**Comment by robertklep** [comment]
Just to be sure, just run 3.0.15 manually from the CLI and it should be fine. No need to reboot.

**Comment by kmarosi** [comment]
@robertklep Unfortunately still the same. :(

**Comment by robertklep** [comment]
Can you check the device capabilities from https://developer.athom.com/tools/devices and see if those match up with the actual state of the curtains?

**Comment by kmarosi** [comment]
@robertklep When I start with the curtains closed and then I click Open in the Home app the curtains open but the values in the Dev tools is not changing. The status remains as Opening... 

Same when I start with them open and I click Close, except the status remains as Closing...

However, if I open/close with Homey the correct status shows up in Home and correct state is shown in the Dev app as well.

**Comment by robertklep** [comment]
@kmarosi if you say that the values in dev tools are not changing, which capabilities specifically aren't changing?

**Comment by kmarosi** [comment]
@robertklep `dim` and `windowcoverings_set`

When I use Homey to open/close `dim` changes.

**Comment by robertklep** [comment]
It doesn't really make sense that they don't change when the curtains change position anyway, because the way it works is that Homeykit changes the `dim` value which is a trigger for the Aqara app to change the curtain position.

So right now I'm at a bit of a loss on why this isn't working for you. Have you tried restarting the Aqara app to see if that solves the problem?

**Comment by kmarosi** [comment]
@robertklep I'm running 4.1.0 on my Homey. I've restarted the Aqara app and even re-installed the curtains in Homey. It got even worse, somehow. I can (somewhat) control it in Homey. If I say I put it on 50%, it chooses whatever it wants as the `dim` level doesn't change in the Dev app. If I manually change the value of `dim` in the Dev app, it goes exactly where I want it and stays there. Even the status in the Home app becomes correct (Instantly) when I change the value in the Dev app.

So I guess with all these facts that is not a Homeykit problem, but more of a problem with the Aqara app or the Zigbee network.

**Comment by robertklep** [comment]
@kmarosi yeah, it's beginning to sound like that :( I don't know how the curtain device works, but if it's battery-operated, I'd start by replacing the batteries.

**Comment by kmarosi** [comment]
And another problem just appeared. When I change the value manually in the Dev app it now sets my curtain to the right position but the value is not updated. Also get a couple of Network request failed so I'm pretty sure now it's the Zigbee network.

The curtains are powered so can't change the batteries. But like I said, it's either the Aqara app or the Zigbee network. But I find it odd that it worked perfectly until now and suddenly it just started to not work a couple of days ago.

Anyway, thanks for all your help. I'll continue investigating this. :p

**Comment by robertklep** [comment]
It could be that the device is _just_ within reach. Because Zigbee uses (almost) the same frequency as WiFi, it might be something silly like a neighbour installing a new WiFi AP that is now causing enough interference that the curtains are dropping out of range.

Perhaps try adding another Zigbee router device in between Homey and the curtains (like an IKEA repeater).

**Comment by kmarosi** [comment]
For some reason Homey and my two Hue bridges were on the same channel. I changed the Zigbee channel on the Hue bridges and then re-installed on the curtains on a rebooted Homey. I'll give it a few days and see if that does the job.
